engine/table/src/test/java/io/deephaven/engine/table/impl/updateby/TestUpdateByGeneral.java

```diff
@@ -394,4 +396,107 @@ public void testResultColumnOrdering() {
+                DateTimeUtils.parseInstant("2022-03-09T09:06:00.000 NY"),
+                DateTimeUtils.parseInstant("2022-03-09T09:07:00.000 NY"),
+                DateTimeUtils.parseInstant("2022-03-09T09:08:00.000 NY"),
+                DateTimeUtils.parseInstant("2022-03-09T09:09:00.000 NY")
+        };
+
+        // incorrect timestamp column name
+        Assert.assertThrows(NoSuchColumnException.class, () -> {
+            final ColumnHolder<?> instantHolder = instantCol("ts", instantArr);
+            final Table t = TableTools.newTable(instantHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("tsCol", prevTime, postTime, "data"));
+        });
+
+        // int as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            final ColumnHolder<?> tsHolder = intCol("ts", new int[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+            final Table t = TableTools.newTable(tsHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("ts", prevTime, postTime, "data"));
+        });
+
+        // short as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            final ColumnHolder<?> tsHolder = shortCol("ts", new short[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+            final Table t = TableTools.newTable(tsHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("ts", prevTime, postTime, "data"));
+        });
+
+        // byte as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            final ColumnHolder<?> tsHolder = byteCol("ts", new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+            final Table t = TableTools.newTable(tsHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("ts", prevTime, postTime, "data"));
+        });
+
+        // float as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            final ColumnHolder<?> tsHolder = floatCol("ts", new float[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+            final Table t = TableTools.newTable(tsHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("ts", prevTime, postTime, "data"));
+        });
+
+        // double as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
+            final ColumnHolder<?> tsHolder = doubleCol("ts", new double[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+            final Table t = TableTools.newTable(tsHolder, dataHolder);
+            final Table summed =
+                    t.updateBy(UpdateByOperation.RollingSum("ts", prevTime, postTime, "data"));
+        });
+
+        // Object as timestamp
+        Assert.assertThrows(IllegalArgumentException.class, () -> {
```

Comment:
This seems like a bug; we should be able to wrap the instant into a timestamp.

Reply:
Modifed the tests a bit. The difference is the generic Object column vs. a column created as Instant